core/drivers/crypto/caam/caam_jr.c

```diff
@@ -0,0 +1,759 @@
+	 * The HW configuration is 64 bits registers regardless
+	 * the CAAM or CPU addressing mode.
+	 */
+	jr_privdata->paddr_inrings =
+		(uint64_t)virt_to_phys(jr_privdata->inrings);
+	jr_privdata->paddr_outrings =
+		(uint64_t)virt_to_phys(jr_privdata->outrings);
+	if ((!jr_privdata->paddr_inrings) || (!jr_privdata->paddr_outrings)) {
+		JR_TRACE("JR bad queue pointers");
+		retstatus = CAAM_FAILURE;
+		goto end_init;
+	}
+
+	hal_jr_config(jr_privdata->baseaddr, jr_privdata->nbJobs,
+		      jr_privdata->paddr_inrings, jr_privdata->paddr_outrings);
+
+	/*
+	 * Prepare the interrupt handler to secure the interrupt even
+	 * if the interrupt is not used
+	 */
+	jr_privdata->it_handler.it = jr_cfg->it_num;
+	jr_privdata->it_handler.flags = ITRF_TRIGGER_LEVEL;
+	jr_privdata->it_handler.handler = caam_jr_irqhandler;
+	jr_privdata->it_handler.data = jr_privdata;
+
+#ifdef CFG_CRYPTO_DRIVER
+	itr_add(&jr_privdata->it_handler);
+#endif
+	hal_jr_enableIT(jr_privdata->baseaddr);
+
+	retstatus = CAAM_NO_ERROR;
+
+end_init:
+	if (retstatus != CAAM_NO_ERROR)
+		do_jr_free(jr_privdata);
+
+	return retstatus;
+}
+
+/*
+ * Request the CAAM JR to halt.
+ * Stop fetching input queue and wait running job completion.
+ *
+ * Returns:
+ * 0    Job Ring is halted
+ * (-1) Error occurred
+ */
+int caam_jr_halt(void)
+{
+	return hal_jr_halt(jr_privdata->baseaddr);
+}
+
+/*
+ * Request the CAAM JR to flush all job running.
+ *
+ * Returns:
+ * 0    Job Ring is halted
+ * (-1) Error occurred
+ */
+int caam_jr_flush(void)
+{
+	return hal_jr_flush(jr_privdata->baseaddr);
+}
+
+/*
+ * Resume the CAAM JR processing.
+ *
+ * @mode    Power mode to resume from
+ */
+void caam_jr_resume(uint32_t pm_hint)
+{
+	if (pm_hint == PM_HINT_CONTEXT_STATE) {
+#if !(defined(CFG_MX6DL) || defined(CFG_MX6D) || defined(CFG_MX6Q) ||          \
+	defined(CFG_MX6QP))
+
+#ifndef CFG_CRYPTO_DRIVER
+		/*
+		 * In case the CAAM is not used the JR used to
+		 * instantiate the RNG has been released to Non-Secure
+		 * hence, need reconfigur the Secure JR and release
+		 * it after RNG instantiation
+		 */
+		hal_jr_setowner(jr_privdata->ctrladdr, jr_privdata->jroffset,
+				JROWN_ARM_S);
+
+		hal_jr_config(jr_privdata->baseaddr, jr_privdata->nbJobs,
+			      jr_privdata->paddr_inrings,
+			      jr_privdata->paddr_outrings);
+#endif
+		/* Read the current job ring index */
+		jr_privdata->inwrite_index =
+			hal_jr_input_index(jr_privdata->baseaddr);
+		/* Read the current output ring index */
+		jr_privdata->outread_index =
+			hal_jr_output_index(jr_privdata->baseaddr);
+
+		if (caam_rng_instantiation() != CAAM_NO_ERROR)
+			panic();
+#ifndef CFG_CRYPTO_DRIVER
+		hal_jr_setowner(jr_privdata->ctrladdr, jr_privdata->jroffset,
+				JROWN_ARM_NS);
+#endif
+#endif
+	} else
+		hal_jr_resume(jr_privdata->baseaddr);
+}
+
+/*
+ * Forces the completion of all CAAM Job to ensure CAAM is not BUSY.
+ *
+ * Returns:
+ * 0    CAAM is no more busy
+ * (-1) CAAM is still busy
```

Comment:
suggest to use `CAAM_NO_ERROR` and `CAAM_BUSY`. Same for other functions above that could use `CAAM_*` status values.

Reply:
ok, thanks